File: Medium/1456-Maximum Number of Vowels in a Substring of Given Length/Sliding Window.rb
# @param {String} s
# @param {Integer} k
# @return {Integer}
def max_vowels(s, k)
    i = 0
    cnt = 0
    ret = 0

    for j in 0...s.length
        if "aeiou".include?(s[j])
            cnt += 1
        end
        if j - i == k
            if "aeiou".include?(s[i])
                cnt -= 1
            end
            i += 1
        end
        ret = [ret, cnt].max
    end

    return ret
end

File: Problemset/1456-Maximum Number of Vowels in a Substring of Given Length/Sliding Window.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        i = 0
        cnt = 0
        ret = 0

        for j in range(len(s)):
            if s[j] in "aeiou":
                cnt += 1
            if j - i == k:
                if s[i] in "aeiou":
                    cnt -= 1
                i += 1
            ret = max(ret, cnt)

        return ret
